### PR TITLE
[Feature][RayJob] Use Use RayContainerIndex instead of 0 (#1397)

### LINKED ISSUE #1397: [Feature] Use RayContainerIndex instead of 0
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

https://github.com/ray-project/kuberay/pull/1375#pullrequestreview-1611726248

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by rueian**
Use `RayContainerIndex` in `ray-operator/controllers/ray/rayjob_controller.go` to fix #1397

Hi @kevin85421,

I currently leave the `rayjob_controller_unit_test.go` untouched so that we can make sure `RayContainerIndex` is 0. Do you think we should change the usage of `0` to `RayContainerIndex` in `rayjob_controller_unit_test.go` as well?

### CODE REVIEW

**Comment by kevin85421** [comment]
> Do you think we should change the usage of 0 to RayContainerIndex in rayjob_controller_unit_test.go as well?

Yep, this will be great.

**Comment by rueian** [comment]
Updated. BTW, Is there any way for contributors to access the above buildkite result? I can only see this:
image

File: ray-operator/controllers/ray/rayjob_controller_unit_test.go [hunk 1]
```diff
@@ -121,25 +122,25 @@ func TestGetSubmitterTemplate(t *testing.T) {
 	// Test 1: User provided template with command
 	submitterTemplate, err := r.getSubmitterTemplate(rayJobInstanceWithTemplate)
 	assert.NoError(t, err)
-	assert.Equal(t, "user-command", submitterTemplate.Spec.Containers[0].Command[0])
+	assert.Equal(t, "user-command", submitterTemplate.Spec.Containers[common.RayContainerIndex].Command[0])
 
 	// Test 2: User provided template without command
 	rayJobInstanceWithTemplate.Spec.SubmitterPodTemplate.Spec.Containers[0].Command = []string{}
```
Comment (kevin85421) [inline ray-operator/controllers/ray/rayjob_controller_unit_test.go:128]:
```suggestion
	rayJobInstanceWithTemplate.Spec.SubmitterPodTemplate.Spec.Containers[common.RayContainerIndex].Command = []string{}
```

**Review by kevin85421 — APPROVED** [review]
LGTM

**Comment by kevin85421** [comment]
The RayService e2e tests are known to be flaky. It has no relationship with this PR. Merge.

**Comment by kevin85421** [comment]
> Updated. BTW, Is there any way for contributors to access the above buildkite result? I can only see this: image

I will figure this out next week. I am not sure how to get the buildkite's access.

**Pull request merged**